vm_backtrace.c

```diff
@@ -568,9 +573,52 @@ backtrace_each(const rb_execution_context_t *ec,
 
     init(arg, size);
 
-    /* SDR(); */
+    /* If a limited number of frames is requested, scan the VM stack for
+     * ignored frames (iseq without pc).  Then adjust the start for the
+     * backtrace to account for skipped frames.
+     */
+    if (start > 0 && num_frames >= 0 && num_frames < real_size) {
+        ptrdiff_t ignored_frames;
+        bool ignored_frames_before_start = false;
+        for (i=0, j=0, cfp = start_cfp; i<last && j<real_size; i++, j++, cfp = RUBY_VM_NEXT_CONTROL_FRAME(cfp)) {
+            if (cfp->iseq && !cfp->pc) {
+                if (j < start)
+                    ignored_frames_before_start = true;
+                else
+                    i--;
+            }
+        }
+        ignored_frames = j - i;
+
+        if (ignored_frames) {
+            if (ignored_frames_before_start) {
+                /* There were ignored frames before start.  So just decrementing
```

Comment:
```suggestion
                /* There were ignored frames before start. So just decrementing
```

Reply:
You may not be aware of this, but two spaces after a period in documentation is used all over Ruby, it's even more popular than the one space after a period:

```
$ grep '/\*.*\.  [A-Z]' *.c |wc -l
      85
$ grep '/\*.*\. [A-Z]' *.c |wc -l
      76
$ grep -r '#.*\.  [A-Z]' lib | wc -l
    1624
$ grep -r '#.*\. [A-Z]' lib | wc -l
    1231
```

So I don't plan on changing this.